WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m

```diff
@@ -28,15 +32,15 @@ @implementation XCElementSnapshot (FBIsVisible)
 
 - (BOOL)fb_isVisible
 {
-  if (self.frame.size.height <= 0 || self.frame.size.width <= 0) {
+  if (CGRectIsEmpty(self.frame) || !CGRectIntersectsRect(self.frame, self.application.frame)) {
```

Comment:
This does not resolve it fully, but just rather makes it to appear less frequently. There are still some cases when this happens and there is no way to prevent these. For example, existence of some active background thread, that changes the UI, while XCTest tries to get accessibility element attributes.

Also, it seems like there are good news about this issue has been fixed in iOS 10.1, but I personally haven't verified it yet. That is why I would advice to disable visibility attribute appending to XML tree (my other PR) by default. Otherwise this might make xpath search even slower than before, even with my fixes :(

Useful links:
https://forums.developer.apple.com/thread/6437
http://stackoverflow.com/questions/36798116/assertion-failure-ui-testing-failure-failure-fetching-attributes-for-element